LibGfx: Refactor Vulkan code and fix some bugs

cqundefine opened the pull request:
This PR makes VulkanContext and VulkanImage closer to the Ladybird C++ code style which also prevents exposed member variables. I've tried enabling validation layers which actually turned out to be quite helpful and exposed a few bugs. I've left that code there locked behind a debug macro.

rcorsi commented on Libraries/LibGfx/VulkanContext.cpp:
```diff
@@ -1,79 +1,152 @@
-    app_info.apiVersion = api_version;
+    app_info.apiVersion = m_api_version;
+
+#ifdef VULKAN_DEBUG
```
all the "ifdef VULKAN_DEBUG" are in other parts of Ladybird as "if constexpr (VULKAN_DEBUG) {" blocks. In the cpp it should be easy enough, but in the header might be a problem. Not sure what to suggest for this. Just wondering if you thought of this?

cqundefine replied:
In some places it is actually `#if`. I don't it matters that much in the end as both achieve the same and I think consistently using `#if` everywhere is better than using `if constexpr` in some places only as in some places it's not possible to use it.

rcorsi commented:
When `vulkan-validationlayers` package (on Ubuntu) is not present you get the error below that prevents VulkanContext creation.  Is that what you want or might it be better to just print a warning, and allow the VulkanContext creation, for example if in CI running test of All_Debug (currently not merged) that might impact some test that might need the VulkanContext? I don't know what the right thing is, just highlighting it to have you consider it. 

```
13055.044 WebContent(141245): Vulkan context creation failed: Request Vulkan validation layer not supported
13055.063 WebContent(141249): Vulkan context creation failed: Request Vulkan validation layer not supported
```

Also the message should be `Requested Vulkan validation layer not supported`, missing the `ed` at end of Request.

cqundefine commented:
> Hi @cqundefine, I just used the validation layer feature you added to troubleshoot #7207 and it gave me some output. Any chance of you just breaking out only the validation layer feature code in a very specific PR so that at least that change can get merged? I would say it is a very useful debug feature to have and would be a good start to merging things.
> 
> A few things I would add / change if I could 1- Name the debug macro VULKAN_VALIDATION_LAYER_DEBUG 2- When a validation layer issue is triggered, dump a backtrace or at least the file and line origin. 3- Maybe this is too much, but item 2 from this list could be controlled with another debug macro VULKAN_VALIDATION_LAYER_BACKTRACE_DEBUG. Just an idea right now, don't know how important that would be.
> 
> If you don't have the time or inclination, would you be okay with me doing it, with you being the author of course and I will be in the `Co-authored-by:`?

I've split of the validation layer part (and only that) in #7293. I did change the macro name to `VULKAN_VALIDATION_LAYER_DEBUG` as it is indeed more descriptive and I've added a call to `dump_backtrace`. I'm not sure a separate macro would make sense for enabling backtraces as if you're using the validation layer the backtrace will most definitely be also useful. If you have any other feedback regarding that feel free to comment on that PR!
